### ISSUE TITLE
Zotero 7 support?

### ISSUE THREAD

**Issue by panangam** [e-zz/logseq-pdf-extract#15]
I'm using Zotero 7 (beta) which seems to not support the "ZotServer" plugin. I assume this will prevent the Logseq plugin from working. Is there any plan to support Zotero 7, or any alternative?

**Comment by e-zz** [comment]
Hi,
The upstream repo of ZotServer has not been updated since the release of Zotero 7. Though not gurantteed, I will look into it when I have time. Thanks for the question!

**Comment by panangam** [comment]
Understandable. Thank you for the response!

**Comment by yeruoforever** [comment]
Thank you for your excellent work.
Does Zotero 7 have its own internal HTTP server that can be used to ensure compatibility with features? What is the necessity of using `ZotServer`?
Can the built-in server be used to ensure that this plugin works?

**Comment by pkuqiuning** [comment]
The author of ZotServer switched to the Local API server in Zotero 7.

https://github.com/MunGell/ZotServer/issues/5#issuecomment-2169426991

**Comment by panangam** [comment]
> MunGell switched to Local API server in Zotero 7.
> 
> [MunGell/ZotServer#5 (comment)](https://github.com/MunGell/ZotServer/issues/5#issuecomment-2169426991)

Oh cool! Does this mean going forward we won't need ZotServer anymore? This means this plugin will need to be modified to use this api instead, right? If that's the case I'll make a new feature request.

**Comment by q-wertz** [comment]
Zotero 7 has been released  :)
https://www.zotero.org/blog/zotero-7/

**Comment by pkuqiuning** [comment]
> > MunGell switched to Local API server in Zotero 7.
> > [MunGell/ZotServer#5 (comment)](https://github.com/MunGell/ZotServer/issues/5#issuecomment-2169426991)
> 
> Oh cool! Does this mean going forward we won't need ZotServer anymore? This means this plugin will need to be modified to use this api instead, right? If that's the case I'll make a new feature request.

Yes, that seems to be the case. The Zotero 7 local API offers more functionality, and it appears that ZotServer has no plans to support Zotero 7.

**Comment by q-wertz** [comment]
Issue should maybe be opened again :see_no_evil:

**Comment by q-wertz** [comment]
Hi @e-zz .
Are there plans to update this project to support Zotero 7 or is this impossible/no time?

**Comment by l8l** [comment]
Hey, I ran into the same issue. ZotServer is not compatible with Zotero 7. At the same time, ZotServer is not continued because there is now a local API that allows queries. This means that the PDF Extract plugin should switch to use the Zotero Local API. Hopefully this won't be too much work? What happens if one sends queries to `http://localhost:23119/search` instead of `http://localhost:23119/zotserver/search` ?

**Comment by e-zz** [comment]
> Hey, I ran into the same issue. ZotServer is not compatible with Zotero 7. At the same time, ZotServer is not continued because there is now a local API that allows queries. This means that the PDF Extract plugin should switch to use the Zotero Local API. Hopefully this won't be too much work? What happens if one sends queries to `http://localhost:23119/search` instead of `http://localhost:23119/zotserver/search` ?

Hi, it's indeed mainly a change in api. I tried the Zotero 7 Local API today and understood a bit how it works. One thing unclear yet to me is data like `citationKey` generated by the Zotero Better BibTeX plugin is not

**Comment by e-zz** [comment]
Hi all, 

Please update to the latest version https://github.com/e-zz/logseq-pdf-extract/releases/tag/0.2.0 to enable Zotero 7 support. 
> Note: You might have to manually select "Zotero 7" in the plugin menu and reload the plugin to make it work.
![Image](https://github.com/user-attachments/assets/2d3f2ddb-db9b-4b28-a83c-05400c6cbd6f)

For any problems you find related to Zotero 7 support, please feel free to open new issues.

Cheers,
